app/models/task_queue.rb

```diff
@@ -0,0 +1,53 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# Used to enqueue tasks that should be run once.
+# The pattern is something like:
+#
+# Initializer - inject class into a list
+# iterate over the list in the cron job
+# If there is a record for that class in the table with no queued_at, then queue a job (and mark it queued)
+# To re-run a task, insert a new row with a blank queued_at value
+class TaskQueue < ApplicationRecord
+  # Marking something as inactive indicates that it should be re-queued
+  scope :active, -> do
+    where(active: true)
+  end
+
+  scope :queued, -> do
+    where(queued_at: nil)
+  end
+
+  # The expectation is that if the task is in available_tasks, then it should be in the queue
+  def self.queue_unprocessed!
+    done = active.pluck(:task_key, :queued_at).to_h
+    available_tasks.each do |task_key, _|
+      next if done[task_key].present?
+
+      t = TaskQueue.create(task_key: task_key)
+      t.delay.run!
+      t.update(queued_at: Time.current)
```

Comment:
Should this maybe mark `active` as well?
```suggestion
      t.update(queued_at: Time.current, active: true)
```

Reply:
Sure, though true is the default in the db.